### PR TITLE
Cacheurl to cachekey - mid qstring drop and light refactor

### PR DESCRIPTION
**Pull request by ezelkow1**
PR 1588 addressed the edges using qstring_drop=1, this handles the mids. Also moved the ats version checking and remap line generation in to functions

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: traffic_ops/app/lib/UI/ConfigFiles.pm [hunk 1]
```diff
@@ -160,6 +160,37 @@ sub server_data {
 	return $server;
 }
 
+sub get_ats_major_version {
```
Comment (alficles) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: This appears to be copy-pasted. Is there no reasonable way to use the same function?

File: traffic_ops/app/lib/UI/ConfigFiles.pm [hunk 2]
```diff
@@ -160,6 +160,37 @@ sub server_data {
...
+		$self->db->resultset('ProfileParameter')
+		->search( { 'parameter.name' => 'trafficserver', 'parameter.config_file' => 'package', 'profile.id' => $server->profile->id },
+		{ prefetch => [ 'profile', 'parameter' ] } )->get_column('parameter.value')->single();
+		
+	if (!defined $ats_ver) {
+	        $ats_ver = "5";
+            $self->app->log->error("Parameter package.trafficserver missing for profile ".$server->profile->name . ". Assuming version $ats_ver");
+        }
+
+	my $ats_major_version = substr( $ats_ver, 0, 1 );
```
Comment (alficles) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: Is this going to fail disastrously at version 10? We're at 8 currently.

File: traffic_ops/app/lib/UI/ConfigFiles.pm [hunk 3]
```diff
@@ -160,6 +160,37 @@ sub server_data {
...
+
+	return $ats_major_version;
+}
+
+sub get_qstring_ignore_remap {
```
Comment (alficles) [inline traffic_ops/app/lib/UI/ConfigFiles.pm]: More copy-paste?

File: traffic_ops/app/lib/UI/ConfigFiles.pm (hunk 2 above)
Comment (ezelkow1) [reply traffic_ops/app/lib/UI/ConfigFiles.pm]: Possibly, but we use this check elsewhere. So when we get to that point its going to fail in a lot of places, not just here

**Comment by ezelkow1** [comment]
Moved the functions to UI::DeliveryService since both configfiles and ats.pm use it

**Comment by ezelkow1** [comment]
this is going to need more changes in the ats.pm since its still referencing server and not server_obj

**Pull request merged**